CI: Travis: Unit test panic in TestResource_WithFakeClient

## CI failure

https://app.travis-ci.com/github/cilium/cilium/jobs/593210550

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   PANIC  package: github.com/cilium/cilium/pkg/k8s/resource • TestResource_WithFakeClient   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
panic: runtime error: invalid memory address or nil pointer dereference [recovered]            
	panic: runtime error: invalid memory address or nil pointer dereference                        
[signal SIGSEGV: segmentation violation code=0x1 addr=0x188 pc=0x1a6f9cd]                      
                                                                                               
goroutine 17 [running]:                                                                        
testing.tRunner.func1.2({0x1c8ce40, 0x31c3350})                                                
	/home/travis/.gimme/versions/go1.19.4.linux.amd64/src/testing/testing.go:1396 +0x24e           
testing.tRunner.func1()                                                                        
	/home/travis/.gimme/versions/go1.19.4.linux.amd64/src/testing/testing.go:1399 +0x39f           
panic({0x1c8ce40, 0x31c3350})                                                                  
	/home/travis/.gimme/versions/go1.19.4.linux.amd64/src/runtime/panic.go:884 +0x212              
github.com/cilium/cilium/pkg/k8s/resource_test.TestResource_WithFakeClient(0xc0004504e0)       
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/k8s/resource/resource_test.go:158 +0xbed  
testing.tRunner(0xc0004504e0, 0x20393f0)                                                       
	/home/travis/.gimme/versions/go1.19.4.linux.amd64/src/testing/testing.go:1446 +0x10b           
created by testing.(*T).Run                                                                    
	/home/travis/.gimme/versions/go1.19.4.linux.amd64/src/testing/testing.go:1493 +0x35f           
FAIL	github.com/cilium/cilium/pkg/k8s/resource	0.162s           
```

--- a/pkg/k8s/resource/resource_test.go
+++ b/pkg/k8s/resource/resource_test.go
@@ -133,7 +133,16 @@ func TestResource_WithFakeClient(t *testing.T) {
 	ev.Done(nil)
 
 	// Second should be a sync.
+	//
+	// We work around the rare race condition in which we see the same
+	// upsert event twice due to it being inserted into store but our
+	// Add handler finishes after the initial listing has been processed (#23079).
+	// Proper fix is to make sure updates to store happen synchronously with queueing
+	// and subscribing.
 	ev = <-events
+	if ev.Kind == resource.Upsert {
+		ev = <-events
+	}
 	assert.Equal(t, resource.Sync, ev.Kind)
 	assert.Nil(t, ev.Object)
 	ev.Done(nil)

